apps/synapse-oauth-signin/src/OAuth2Form.tsx

```diff
@@ -60,248 +73,192 @@ export function OAuth2Form() {
     [clearSession],
   )
 
-  // In addition to fetching the current user profile, the success of this request will determine if the current access token is valid.
-  const { data: profile, error: fetchProfileError } =
-    SynapseQueries.useGetCurrentUserProfile({
-      enabled: !!accessToken,
+  const clientId = useMemo(() => {
+    const code = queryParams.get('code')
+    if (code) return // we're in the middle of a SSO, do not attempt to get OAuthClient info yet
+
+    const clientId = queryParams.get('client_id')
+    if (!clientId) {
+      onError(new Error('Synapse OAuth client_id is required'))
+      return
+    }
+    return clientId
+  }, [onError, queryParams])
+
+  const { data: oauthClientInfo, isLoading: isLoadingClientInfo } =
+    SynapseQueries.useGetOAuth2Client(clientId!, {
+      enabled: Boolean(clientId),
     })
 
+  const history = useHistory()
+
+  // In addition to fetching the current user profile, the success of this request will determine if the current access token is valid.
+  const {
+    data: profile,
+    error: fetchProfileError,
+    isLoading: isLoadingProfile,
+  } = SynapseQueries.useGetCurrentUserProfile({
+    enabled: isLoggedIn,
+  })
+
   useEffect(() => {
     if (fetchProfileError) {
       onError(fetchProfileError)
     }
   }, [fetchProfileError, onError])
 
   if (profile?.profilePicureFileHandleId) {
+    // Note: `getPortalFileHandleServletUrl` is not a web request.
     profile.clientPreSignedURL = SynapseClient.getPortalFileHandleServletUrl(
       profile.profilePicureFileHandleId,
       profile.ownerId,
       FileHandleAssociateType.UserProfileAttachment,
     )
   }
 
-  const [oidcRequestDescription, setOidcRequestDescription] =
-    useState<OIDCAuthorizationRequestDescription>()
-  const [oauthClientInfo, setOauthClientInfo] = useState<OAuthClientPublic>()
-  const [redirectURL, setRedirectURL] = useState<string>()
-  const [isPreviousAuthCheckComplete, setIsPreviousAuthCheckComplete] =
-    useState<boolean>(false)
-  const [isConsenting, setIsConsenting] = useState<boolean>(false)
```

Comment:
In general, most of the old state variables are now

1. Derived from a react-query query or mutation,
2. Derived within a `useMemo`, or
3. Gone because they were unnecessary